### PR TITLE
Setup documentation via sphinx

### PR DESCRIPTION
**Pull request by francescofarina**
This PR:
- set up the docs (https://github.com/TristanBilot/mlx-graphs/issues/19)
- refactors some docstrings and files to make them compatible with sphinx
- adds a PR template to ensure we write tests and docs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,3 @@
-# mlx-graphs specific ignores
-data/
```
Comment (TristanBilot) [inline .gitignore:2]: I put this one as GCN and GAT examples download the Cora dataset into a data/ folder, and it's cleaner to add this one to .gitignore.

File: docs/make.bat
Comment (TristanBilot) [inline docs/make.bat:1]: I never used sphinx, but is it normal to push a `bat` file to the repo? Isn't it automatically generated?

**Review by TristanBilot — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (francescofarina) [reply .gitignore:2]: > I put this one as GCN and GAT examples download the Cora dataset into a data/ folder, and it's cleaner to add this one to .gitignore.

Yeah, the problem is that we have modules and files with the `data` pattern that would be gitignored with that line. I suggest renaming the dataset folder to something different (possibly a dotfile).

File: docs/make.bat
Comment (francescofarina) [reply docs/make.bat:1]: Good catch, I'll remove it before merging!

**Comment by TristanBilot** [comment]
Very nice work btw, it's very clean :)

**Comment by TristanBilot** [comment]
I have some errors when doing the `make html`, do you also have them?
```
WARNING: [autosummary] failed to import mlx_graphs.nn.GATConv.
Possible hints:
* ModuleNotFoundError: No module named 'mlx_graphs.nn.GATConv'
* ImportError: 
* AttributeError: module 'mlx_graphs.nn' has no attribute 'GATConv'
WARNING: [autosummary] failed to import mlx_graphs.nn.GCNConv.
Possible hints:
* AttributeError: module 'mlx_graphs.nn' has no attribute 'GCNConv'
* ImportError: 
* ModuleNotFoundError: No module named 'mlx_graphs.nn.GCNConv'
* ```

**Comment by TristanBilot** [comment]
Much better, but still have these ones:

```
WARNING: [autosummary] failed to import mlx_graphs.nn.GraphNetworkBlock.
Possible hints:
* ModuleNotFoundError: No module named 'mlx_graphs.nn.GraphNetworkBlock'
* AttributeError: module 'mlx_graphs.nn' has no attribute 'GraphNetworkBlock'
* ImportError: 
WARNING: [autosummary] failed to import mlx_graphs.nn.Linear.
Possible hints:
* ImportError: 
* ModuleNotFoundError: No module named 'mlx_graphs.nn.Linear'
* AttributeError: module 'mlx_graphs.nn' has no attribute 'Linear'
WARNING: [autosummary] failed to import mlx_graphs.nn.MessagePassing.
Possible hints:
* AttributeError: module 'mlx_graphs.nn' has no attribute 'MessagePassing'
* ModuleNotFoundError: No module named 'mlx_graphs.nn.MessagePassing'
* ImportError:
```

**Comment by francescofarina** [comment]
> Much better, but still have these ones:
> 
> ```
> WARNING: [autosummary] failed to import mlx_graphs.nn.GraphNetworkBlock.
> Possible hints:
> * ModuleNotFoundError: No module named 'mlx_graphs.nn.GraphNetworkBlock'
> * AttributeError: module 'mlx_graphs.nn' has no attribute 'GraphNetworkBlock'
> * ImportError: 
> WARNING: [autosummary] failed to import mlx_graphs.nn.Linear.
> Possible hints:
> * ImportError: 
> * ModuleNotFoundError: No module named 'mlx_graphs.nn.Linear'
> * AttributeError: module 'mlx_graphs.nn' has no attribute 'Linear'
> WARNING: [autosummary] failed to import mlx_graphs.nn.MessagePassing.
> Possible hints:
> * AttributeError: module 'mlx_graphs.nn' has no attribute 'MessagePassing'
> * ModuleNotFoundError: No module named 'mlx_graphs.nn.MessagePassing'
> * ImportError:
> ```

@TristanBilot can you retry? Also try to `rm -r build` before building.

**Comment by TristanBilot** [comment]
It works now fine on my side

**Pull request merged**